elastic/elastic-package
# CLI to install a package

Fixes: https://github.com/elastic/elastic-package/issues/87
Fixes: https://github.com/elastic/elastic-package/issues/8

This PR exposes CLI action to install the current package (work dir), available in the Package Registry.

It introduces also the "check package conditions" feature to verify if the package meets Elastic stack conditions. It will be used by https://github.com/elastic/integrations/issues/537#issuecomment-779253207 to check in advance whether the stack can expose the package.

How to test the PR:

1. Go to the `test/packages/kubernetes`.
2. `elastic-package build`
3. `elastic-package stack up -d` (to include the package)
4. `elastic-package install` (to install the package)

Check constraints (Kibana is not required to be up):

```
➜  apache git:(87-install-package) elastic-package install -c kibana.version=7.8.0 -v
2021/02/16 13:00:00 DEBUG Enable verbose logging
Check conditions for package
2021/02/16 13:00:00 DEBUG Verify Kibana version (constraint: ^7.9.0, requirement: 7.8.0)
Error: checking conditions failed: Kibana constraint unsatisfied: [0] 7.8.0 does not have same major version as 7.9.0
➜  apache git:(87-install-package) elastic-package install -c kibana.version=7.11.0 -v
2021/02/16 13:00:04 DEBUG Enable verbose logging
Check conditions for package
2021/02/16 13:00:04 DEBUG Verify Kibana version (constraint: ^7.9.0, requirement: 7.11.0)
2021/02/16 13:00:04 DEBUG Constraint kibana.version = ^7.9.0 is satisfied
Done
```
